### ISSUE TITLE
Read performance heavily degraded on ZFS volume

### ISSUE THREAD

**Issue by in4mer** [openzfs/zfs#1049]
Hello,

I have five 2tb drives in a single raidz pool.

One of the filesystems, grandchild of the root filesystem, has about 800GB in it.  They're larger files, between 5 and 50GB.  Most of the files were written at about 2MBps+.

For the most part, I haven't been paying attention to the read speeds.  If it's above a certain (absurdly low) threshold, I don't notice performance problems.  It has come to my attention, however, that these minimal performance requirements can no longer be met.

This was a dd on a single file earlier today:

dd bs=4096 if=/lfs/media/my_username/file.mkv of=./r-s.mkv
1429826+1 records in
1429826+1 records out
5856569821 bytes (5.9 GB) copied, 13366 s, 438 kB/s

The drive being copied to runs ext4, and was unloaded.  I was running iostat in the background while the dd was running.  60 second interval.  Here's a random slice of its output from near the end (stuff near the beginning had a MUCH higher individual disk read value on each member of the zfs pool).  The last five disks are the pool members, and everything before are conventional filesystems:

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
sda               0.00    23.57    0.78   29.27     7.33   287.33    19.61     0.44   14.66   11.57   14.74   7.96  23.93
sdb               0.00     1.03    0.07    6.62     0.27  2651.73   793.62     1.27  190.07   60.00  191.39   4.58   3.06
md0               0.00     0.00    0.00    1.12     0.00     4.47     8.00     0.00    0.00    0.00    0.00   0.00   0.00
dm-0              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
dm-1              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
dm-2              0.00     0.00    0.00    1.02     0.00     4.47     8.79     0.01    9.51    0.00    9.51   7.54   0.77
dm-3              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
dm-4              0.00     0.00    0.78   47.87     7.33   281.80    11.89     0.49   10.02   11.57    9.99   4.82  23.44
dm-5              0.00     0.00    0.07    6.07     0.27  2646.20   862.98     1.28  208.65   60.00  210.29   4.01   2.46
dm-6              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
sdc               0.00     0.00 2121.77    0.00 15398.35     0.00    14.51     0.45    0.21    0.21    0.00   0.21  44.55
sdg               0.00     0.00 1590.78    0.00 18010.08     0.00    22.64     0.39    0.25    0.25    0.00   0.25  39.12
sde               0.00     0.00 1660.23    0.00 18506.92     0.00    22.29     0.38    0.23    0.23    0.00   0.23  38.21
sdd               0.00     0.00 2164.00    0.00 16033.07     0.00    14.82     0.44    0.20    0.20    0.00   0.20  43.80
sdf               0.00     0.00 1039.10    0.00 18734.72     0.00    36.06     0.23    0.22    0.22    0.00   0.22  22.99

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           6.44    0.00    5.18    0.73    0.00   87.65

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
sda               0.00    12.47    0.00   16.23     0.00   296.53    36.53     0.17   10.25    0.00   10.25   7.17  11.65
sdb               0.00     0.67    0.07    3.40     0.27  1136.27   655.69     0.46  132.81   63.00  134.18   4.40   1.53
md0               0.00     0.00    0.00    0.70     0.00     2.67     7.62     0.00    0.00    0.00    0.00   0.00   0.00
dm-0              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
dm-1              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
dm-2              0.00     0.00    0.00    0.63     0.00     2.67     8.42     0.00    6.95    0.00    6.95   5.79   0.37
dm-3              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
dm-4              0.00     0.00    0.00   25.95     0.00   293.20    22.60     0.18    6.93    0.00    6.93   4.39  11.39
dm-5              0.00     0.00    0.07    3.00     0.27  1132.93   739.04     0.46  149.96   63.00  151.89   3.85   1.18
dm-6              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
sdc               0.00     0.00 2120.17    0.00 17816.52     0.00    16.81     0.47    0.22    0.22    0.00   0.22  46.73
sdg               0.00     0.00 1504.42    0.00 16141.33     0.00    21.46     0.36    0.24    0.24    0.00   0.24  36.06
sde               0.00     0.00 1493.10    0.00 16024.12     0.00    21.46     0.37    0.25    0.25    0.00   0.25  36.83
sdd               0.00     0.00 1992.72    0.00 15819.23     0.00    15.88     0.41    0.21    0.21    0.00   0.21  41.12
sdf               0.00     0.00  776.72    0.00 13920.18     0.00    35.84     0.16    0.20    0.20    0.00   0.20  15.62

In a nutshell, it says that each member of the zfs pool was reading an average of 15944 kBps read (79.7MBps aggregate read speed), while the aggregate _write_ speed was 1136kBps to the ext4 filesystem.

This is a difficulty.

How can I provide more information to make it possible to run this problem down?  I am at your service.
